### PR TITLE
[WIP] Some bug fixes.

### PR DESCRIPTION
**Pull request by jorive**
- Fix broken glob introduced with #341.
- Attempt to read commit sha from core-sdk (SDK) instead of core-setup (Runtime). This change is required to report accurately the .NET product being tested.

Fixes https://github.com/dotnet/performance/issues/346
Fixes https://github.com/dotnet/performance/issues/347

### CODE REVIEW

**Comment by adamsitnik** [comment]
The Ubuntu runs fail with:

```log
Traceback (most recent call last):
  File "scripts/benchmarks_ci.py", line 250, in <module>
    __main(sys.argv[1:])
  File "scripts/benchmarks_ci.py", line 245, in __main
    benchview.run_scripts(args, verbose, BENCHMARKS_CSPROJ)
  File "/home/helixbot/dotnetbuild/work/fc1c2619-4f89-4f3d-9694-3fe511271749/Work/95b092f5-0f33-4ae7-afe2-4417e5642667/Exec/scripts/benchview.py", line 452, in run_scripts
    __run_scripts(args, benchviewpy, framework)
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/helixbot/dotnetbuild/work/fc1c2619-4f89-4f3d-9694-3fe511271749/Work/95b092f5-0f33-4ae7-afe2-4417e5642667/Exec/scripts/performance/common.py", line 105, in push_dir
    yield
  File "/home/helixbot/dotnetbuild/work/fc1c2619-4f89-4f3d-9694-3fe511271749/Work/95b092f5-0f33-4ae7-afe2-4417e5642667/Exec/scripts/benchview.py", line 452, in run_scripts
    __run_scripts(args, benchviewpy, framework)
  File "/home/helixbot/dotnetbuild/work/fc1c2619-4f89-4f3d-9694-3fe511271749/Work/95b092f5-0f33-4ae7-afe2-4417e5642667/Exec/scripts/benchview.py", line 391, in __run_scripts
    buildinfo = __get_build_info(args, target_framework_moniker)
  File "/home/helixbot/dotnetbuild/work/fc1c2619-4f89-4f3d-9694-3fe511271749/Work/95b092f5-0f33-4ae7-afe2-4417e5642667/Exec/scripts/benchview.py", line 302, in __get_build_info
    which('dotnet')
  File "/home/helixbot/dotnetbuild/work/fc1c2619-4f89-4f3d-9694-3fe511271749/Work/95b092f5-0f33-4ae7-afe2-4417e5642667/Exec/scripts/dotnet.py", line 317, in get_dotnet_sdk
    'Did not find "Base Path:" entry on the `dotnet --info` command'
RuntimeError: Did not find "Base Path:" entry on the `dotnet --info` command
```

It's weird because in the logs we can find the output of `dotnet --info` and "Base Path" is there:

```log
[2019/03/04 06:37:30][INFO] $ dotnet --info
[2019/03/04 06:37:31][INFO] .NET Core SDK (reflecting any global.json):
[2019/03/04 06:37:31][INFO]  Version:   3.0.100-preview4-010591
[2019/03/04 06:37:31][INFO]  Commit:    6cb32d53bc
[2019/03/04 06:37:31][INFO] 
[2019/03/04 06:37:31][INFO] Runtime Environment:
[2019/03/04 06:37:31][INFO]  OS Name:     ubuntu
[2019/03/04 06:37:31][INFO]  OS Version:  16.04
[2019/03/04 06:37:31][INFO]  OS Platform: Linux
[2019/03/04 06:37:31][INFO]  RID:         ubuntu.16.04-x64
[2019/03/04 06:37:31][INFO]  Base Path:   /home/helixbot/dotnetbuild/work/fc1c2619-4f89-4f3d-9694-3fe511271749/Work/95b092f5-0f33-4ae7-afe2-4417e5642667/Exec/tools/dotnet/x64/sdk/3.0.100-preview4-010591/
```

File: scripts/benchview.py [hunk 1]
```diff
@@ -404,64 +376,80 @@ def run_scripts(
...
-            )
+    # BenchView submission-metadata.py
+    submission_name = '%s (%s): %s:%s' % (
+        args.benchview_submission_name,
+        args.benchview_run_type,
+        urlparse(buildinfo.repository).path,
+        buildinfo.commit_sha
+    )
+
+    # BenchView submission-metadada.py
```
[the remark this answers is not in the document]
Comment (jorive) [reply scripts/benchview.py]: Sorry, I will remove it... I add those as a mock of what I need to do.

**Comment by jorive** [comment]
@adamsitnik Taking a look at the failure now.

**Comment by jorive** [comment]
@adamsitnik I'm unable to repro that failure in two different `Ubuntu 18.04` machines. Instead, I am getting this error:
```log
[2019/03/04 11:36:37][INFO] --------------------------------------------------
[2019/03/04 11:36:37][INFO] Dumping COMPlus environment:
[2019/03/04 11:36:37][INFO]   "COMPlus_TieredCompilation=1"
[2019/03/04 11:36:37][INFO] --------------------------------------------------
[2019/03/04 11:36:37][INFO] $ pushd "/home/jorive/src/github/jorive/performance/src/benchmarks/micro"
[2019/03/04 11:36:37][INFO] $ dotnet run --project /home/jorive/src/github/jorive/performance/src/benchmarks/micro/MicroBenchmarks.csproj --configuration Release --framework netcoreapp2.1 --no-restore --no-build -- --filter *Adams* --packages /home/jorive/src/github/jorive/performance/artifacts/packages --runtimes netcoreapp2.1
[2019/03/04 11:36:39][INFO]
[2019/03/04 11:36:39][INFO] Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: The type initializer for 'System.Drawing.Tests.Perf_Image_Load' threw an exception. ---> System.TypeInitializationException: The type initializer for 'Gdip' threw an exception. ---> System.DllNotFoundException: Unable to load DLL 'libgdiplus': The specified module could not be found.
[2019/03/04 11:36:39][INFO]    at System.Runtime.InteropServices.FunctionWrapper`1.get_Delegate()
[2019/03/04 11:36:39][INFO]    at System.Drawing.SafeNativeMethods.Gdip.GdiplusStartup(IntPtr& token, StartupInput& input, StartupOutput& output)
[2019/03/04 11:36:39][INFO]    at System.Drawing.SafeNativeMethods.Gdip..cctor()
[2019/03/04 11:36:39][INFO]    --- End of inner exception stack trace ---
[2019/03/04 11:36:39][INFO]    at System.Drawing.SafeNativeMethods.Gdip.GdipCreateBitmapFromScan0(Int32 width, Int32 height, Int32 stride, Int32 format, HandleRef scan0, IntPtr& bitmap)
[2019/03/04 11:36:39][INFO]    at System.Drawing.Bitmap..ctor(Int32 width, Int32 height, PixelFormat format)
[2019/03/04 11:36:39][INFO]    at System.Drawing.Bitmap..ctor(Int32 width, Int32 height)
[2019/03/04 11:36:39][INFO]    at System.Drawing.Tests.Perf_Image_Load.ImageTestData.CreateTestImage(ImageFormat format) in /home/jorive/src/github/jorive/performance/src/benchmarks/micro/corefx/System.Drawing/Perf_Image_Load.cs:line 76
[2019/03/04 11:36:39][INFO]    at System.Drawing.Tests.Perf_Image_Load.ImageTestData..ctor(ImageFormat format) in /home/jorive/src/github/jorive/performance/src/benchmarks/micro/corefx/System.Drawing/Perf_Image_Load.cs:line 60
[2019/03/04 11:36:39][INFO]    at System.Drawing.Tests.Perf_Image_Load..cctor() in /home/jorive/src/github/jorive/performance/src/benchmarks/micro/corefx/System.Drawing/Perf_Image_Load.cs:line 17
[2019/03/04 11:36:39][INFO]    --- End of inner exception stack trace ---
[2019/03/04 11:36:39][INFO]    at System.Drawing.Tests.Perf_Image_Load.ImageFormats() in /home/jorive/src/github/jorive/performance/src/benchmarks/micro/corefx/System.Drawing/Perf_Image_Load.cs:line 24
[2019/03/04 11:36:39][INFO]    --- End of inner exception stack trace ---
[2019/03/04 11:36:39][INFO]    at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
[2019/03/04 11:36:39][INFO]    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.BenchmarkConverter.GetValidValuesForParamsSource(Type parentType, String sourceName)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.BenchmarkConverter.GetArgumentsDefinitions(MethodInfo benchmark, Type target)+MoveNext()
[2019/03/04 11:36:39][INFO]    at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
[2019/03/04 11:36:39][INFO]    at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
[2019/03/04 11:36:39][INFO]    at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.BenchmarkConverter.MethodsToBenchmarksWithFullConfig(Type containingType, MethodInfo[] benchmarkMethods, ImmutableConfig immutableConfig)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.BenchmarkConverter.TypeToBenchmarks(Type type, IConfig config)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.TypeFilter.<>c__DisplayClass1_0.<Filter>b__0(Type type)
[2019/03/04 11:36:39][INFO]    at System.Linq.Enumerable.SelectListIterator`2.MoveNext()
[2019/03/04 11:36:39][INFO]    at System.Linq.Enumerable.WhereEnumerableIterator`1.ToArray()
[2019/03/04 11:36:39][INFO]    at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.TypeFilter.Filter(IConfig effectiveConfig, IEnumerable`1 types)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.BenchmarkSwitcher.RunWithDirtyAssemblyResolveHelper(String[] args, IConfig config)
[2019/03/04 11:36:39][INFO]    at BenchmarkDotNet.Running.BenchmarkSwitcher.Run(String[] args, IConfig config)
[2019/03/04 11:36:39][INFO]    at MicroBenchmarks.Program.Main(String[] args) in /home/jorive/src/github/jorive/performance/src/benchmarks/micro/Program.cs:line 15
[2019/03/04 11:36:39][ERROR] Process exited with status 134
[2019/03/04 11:36:39][INFO] $ popd
Traceback (most recent call last):
  File "./scripts/benchmarks_ci.py", line 250, in <module>
    __main(sys.argv[1:])
  File "./scripts/benchmarks_ci.py", line 242, in __main
    args
  File "/home/jorive/src/github/jorive/performance/scripts/micro_benchmarks.py", line 371, in run
    *run_args
  File "/home/jorive/src/github/jorive/performance/scripts/dotnet.py", line 284, in run
    self.working_directory)
  File "/home/jorive/src/github/jorive/performance/scripts/performance/common.py", line 185, in run
    proc.returncode, quoted_cmdline)
subprocess.CalledProcessError: Command '$ dotnet run --project /home/jorive/src/github/jorive/performance/src/benchmarks/micro/MicroBenchmarks.csproj --configuration Release --framework netcoreapp2.1 --no-restore --no-build -- --filter *Adams* --packages /home/jorive/src/github/jorive/performance/artifacts/packages --runtimes netcoreapp2.1' returned non-zero exit status 134.
```

Have you seen it? Any idea?

**Comment by jorive** [comment]
The `.version` commit sha repository changed for `netcoreapp3.0`.
Currently the commit sha found in the `.version` file corresponds to `dotnet/core-sdk` repository, when previously it used to be `dotnet/cli`.

**Comment by jorive** [comment]
@adamsitnik Could you please take another look?

**Comment by jorive** [comment]
On `__get_build_info` we are able to determine the build information of the .NET Cli tools that were downloaded by inspecting the SDK information. What should we use for `net461`? Is there some information that can be pulled from the Desktop?

**Comment by adamsitnik** [comment]
@jkotas Jan is there any way to get the commit sha of given Full .NET Framework?

Following code works great for .NET Core, but not for Full Framework:

```cs
var coreClrAssemblyInfo = FileVersionInfo.GetVersionInfo(typeof(object).Assembly.Location);
Console.WriteLine($"{coreClrAssemblyInfo.ProductVersion}");
```

```log
PS C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0
4.6.27121.03 @BuiltBy: dlab14-DDVSOWINAGE101 @Branch: master @SrcCode: https://github.com/dotnet/coreclr/tree/0197d4e9040c3081817d500150beb0f65cbc5672
PS C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp2.2
4.6.27110.04 @BuiltBy: dlab14-DDVSOWINAGE056 @Branch: release/2.2 @SrcCode: https://github.com/dotnet/coreclr/tree/418e0949164eb91e73f7884387bcf9661016ba9f
PS C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp2.1
4.6.27019.06 @BuiltBy: dlab14-DDVSOWINAGE057 @Branch: release/2.1 @SrcCode: https://github.com/dotnet/coreclr/tree/9bff252639252a2fa0517c12b39a1af9390c9ebd
PS C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp2.0
4.6.26614.01 @BuiltBy: dlab14-DDVSOWINAGE018 @Commit: a536e7eec55c538c94639cefe295aa672996bf9b
PS C:\Projects\performance\src\benchmarks\micro> dotnet run -f net461
4.7.3324.0
```

File: scripts/dotnet.py [hunk 2]
```diff
@@ -9,13 +9,15 @@
...
 from logging import getLogger
-from os import chmod, environ, makedirs, path, pathsep
+from os import chmod, environ, listdir, makedirs, path, pathsep
 from stat import S_IRWXU
 from subprocess import check_output
 from sys import argv, platform
 from urllib.parse import urlparse
 from urllib.request import urlopen, urlretrieve
 
+import re
```
Comment (adamsitnik) [inline scripts/dotnet.py]: not related to the PR but to the language: python named the regex module just `re`? that's crazy

**Comment by jkotas** [comment]
> Jan is there any way to get the commit sha of given Full .NET Framework?

.NET Framework does not live in git. There are no commit hashes for .NET Framework. The FileVersion is the identifier to use for .NET Framework (e.g. `4.7.3324.0 built by: NET472REL1LAST_C`).

**Comment by adamsitnik** [comment]
@jkotas thank you!

@jorive I am going to send a PR with a fix

File: scripts/dotnet.py (hunk 2 above)
Comment (jorive) [reply scripts/dotnet.py]: Yes, agreed :( … how hard it would have been to name it something like: `regex`, `regular_expression`

**Comment by jorive** [comment]
@adamsitnik Can I merge?

**Comment by adamsitnik** [comment]
@jorive  :shipit:

**Pull request merged**